src/app/CommandHandler.h

```diff
@@ -85,6 +85,12 @@ class CommandHandler
          * fails to exist.
          */
         virtual Protocols::InteractionModel::Status CommandExists(const ConcreteCommandPath & aCommandPath) = 0;
+
+        /*
+         * Get the generation number of the InteractionModelEngine. A CommandHandler::Handle is valid iff
+         * its number is equals to the InteractionModelEngine's one.
+         */
+        virtual uint32_t GetInteractionModelEngineGeneration() const = 0;
```

Comment:
This feels off ... why do we expose a magic constant externally? Why not have  "IsValid" call instead if that is the purpose. This seems to leak some implementation details into virtual base classes.

Reply:
The CommandHandler::Handle takes this number during constructure and make sure that it is still valid during `Handle` calls. If this is changed to isValid() calls, InteractionModelEngine will somehow need to store something for this CommandHandler. I don't think that is a better solution. Do you have other suggestions?

Reply:
I also update the function name so that it doesn't sound like an impletation details in ImEngine